dotCMS/src/main/java/com/dotcms/security/apps/AppDescriptorHelper.java

```diff
@@ -0,0 +1,384 @@
+package com.dotcms.security.apps;
+
+import static com.dotcms.security.apps.AppsAPI.APPS_DIR_NAME;
+import static com.dotcms.security.apps.AppsAPI.APPS_DIR_PATH_KEY;
+import static com.dotcms.security.apps.AppsAPI.DESCRIPTOR_NAME_MAX_LENGTH;
+import static com.dotcms.security.apps.AppsAPI.SERVER_DIR_NAME;
+import static com.dotmarketing.util.UtilMethods.isNotSet;
+import static com.dotmarketing.util.UtilMethods.isSet;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotDataValidationException;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.fasterxml.jackson.core.JsonParser.Feature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.collect.ImmutableList;
+import com.liferay.util.StringPool;
+import io.vavr.Tuple;
+import io.vavr.Tuple2;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class AppDescriptorHelper {
+
+    private final ObjectMapper ymlMapper = new ObjectMapper(new YAMLFactory())
+            .enable(Feature.STRICT_DUPLICATE_DETECTION)
+            //.enable(SerializationFeature.INDENT_OUTPUT)
+            .findAndRegisterModules();
+
+    /**
+     * There's a version of the method readValue on the ymlMapper which takes a file and internally creates directly a FileInputStream
+     * According to https://dzone.com/articles/fileinputstream-fileoutputstream-considered-harmful
+     * that's very harmful
+     * @param file
+     * @return
+     * @throws DotDataException
+     */
+    AppSchema readAppFile(final Path file) throws DotDataException {
+        try (InputStream inputStream = Files.newInputStream(file)) {
+            return ymlMapper.readValue(inputStream, AppSchema.class);
+        }catch (Exception e){
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6286472)